pkg/icingaredis/telemetry/liveness.go

```diff
@@ -0,0 +1,24 @@
+package telemetry
+
+import (
+	"context"
+	"github.com/go-redis/redis/v8"
+	"github.com/icinga/icingadb/pkg/icingaredis"
+)
+
+// IndicateLiveness signals to Icinga 2 (for monitoring purpose) that Icinga DB is alive
+// (just by being subscribed to icingadb:telemetry:liveness).
```

Comment:
1. Yes.
2. The triggered health checks... I guess.
3. Well, this approach is realtime.

Reply:
2. Which health check? TCP connections can stay alive for quite some time with one device gone. Put your laptop to sleep for a an hour, wake it up again and if you're lucky and it gets the same address again, there's a good chance that a SSH session started before is still alive. We have to make sure that this can't happen here, if the Redis server does anything that helps us achieving this, fine, but have to make sure it does.

Reply:
https://github.com/Icinga/icingadb/pull/464/files#diff-b5c6a821d24c84fa4a6344723b1a0f00374fddd35e157971ee1f20c34363f168R15

Reply:
Sounds like that part of the documentation for `Channel()`:

> go-redis periodically sends ping messages to test connection health and re-subscribes if ping can not not received for 30 seconds.

That would help for detecting this situation when the Redis server is gone (or the connection broke, for example some stateful firewall in between dropping the connection), but we want to detect the situation where the client (i.e. Icinga DB) is gone, so the Redis server would have to support something similar.

According to the [Redis client handling docs](https://redis.io/docs/reference/clients/#tcp-keepalive) the only thing it does is enabling TCP keepalive by default with an interval of 300 seconds which is longer than what I'd say we want to have as a limit.

Reply:
Then the check should s/PUBSUB NUMSUB/PUBLISH/

Reply:
How does that help? I mean this would probably kill the connection at some time because the send buffer is full, but then the timeout would depend on the check frequency (more checks = more messages = buffer filler earlier).

I don't really see how this is any better than just writing a timestamp, so that we know some point in time where Icinga DB was alive for sure. Which is also what Icinga 2 does, so why not use the same mechanism in both directions?